File: src/FrameIO/Models/ComLogModels.cs
using System;
using static FrameHelper.EntityHelpers;

namespace FrameIO.Models
{
    public interface IComLogDTO : IWithIdAndTimeStamp
    {
        string UserId { get; set; }
        string Port { get; set; }
        string Action { get; set; }
        string Location { get; set; }
    }

    public class ComLogDTO : IComLogDTO
    {
        public Guid Id { get; set; }
        public DateTime TimeStamp { get; set; }
        public string UserId { get; set; }
        public string Port { get; set; }
        public string Action { get; set; }
        public string Location { get; set; }
    }

    public interface IComLogView : IComLogDTO
    {
    }

    public class ComLogView : ComLogDTO, IComLogView
    {
    }

}

File: src/FrameUI/src/app/models/ComLogModels.ts


export interface IcomLogDTO {
    id: string;
    timeStamp: Date;
    userId: string;
    port: string;
    action: string;
    location: string;
    
}

export class ComLogDTO implements IcomLogDTO {
    public id: string;
    
    public timeStamp: Date;
    
    public userId: string;
    
    public port: string;
    
    public action: string;
    
    public location: string;
    
    constructor(model?: IcomLogDTO) {
        if(model) {
            this.id = model.id;
            this.timeStamp = model.timeStamp;
            this.userId = model.userId;
            this.port = model.port;
            this.action = model.action;
            this.location = model.location;
            
        }
    }
}

export interface IcomLogView extends IcomLogDTO {
    
}

export class ComLogView extends ComLogDTO implements IcomLogView {
    constructor(model?: IcomLogView) {
	super(model);

        if(model) {
            
        }
    }
}
 